tests/win_named_pipe.rs

```diff
@@ -0,0 +1,267 @@
+#![cfg(windows)]
```

Comment:
Nit: These tests can use some of the test utilities in `tests/util`, such as `expect_events`, but that can be done later.